src/apprt/gtk/Surface.zig

```diff
@@ -2010,3 +2021,121 @@ pub fn setSplitZoom(self: *Surface, new_split_zoom: bool) void {
 pub fn toggleSplitZoom(self: *Surface) void {
     self.setSplitZoom(!self.zoomed_in);
 }
+
+pub fn ShellEscapeWriter(comptime T: type) type {
```

Comment:
I'd like to see this moved out of the `apprt/gtk` package and somewhere else with a good set of unit tests.

"Somewhere else" is probably `src/terminal`?

Reply:
Yeah I stuck it there for convenience but I agree that it should be elsewhere, maybe `src/os/shell.zig` since it's kinda "shell" related.

Reply:
Agree with that location.